server/src/storage/uncompressed_data_manager.rs

```diff
@@ -129,146 +137,174 @@ impl UncompressedDataManager {
+            if UncompressedInMemoryDataBuffer::get_memory_size()
+                > self.uncompressed_remaining_memory_in_bytes
+            {
+                self.spill_finished_buffer();
             }
 
-            // Create a new segment and reduce the remaining amount of reserved memory by its size.
-            let mut segment = SegmentBuilder::new();
-            self.uncompressed_remaining_memory_in_bytes -= SegmentBuilder::get_memory_size();
+            // Create a new buffer and reduce the remaining amount of reserved memory by its size.
+            let mut buffer = UncompressedInMemoryDataBuffer::new(univariate_id);
+            self.uncompressed_remaining_memory_in_bytes -=
+                UncompressedInMemoryDataBuffer::get_memory_size();
 
             debug!(
                 "Created segment. Remaining reserved bytes: {}.",
                 self.uncompressed_remaining_memory_in_bytes
             );
 
-            segment.insert_data(timestamp, value);
-            self.uncompressed_data.insert(key, segment);
+            buffer.insert_data(timestamp, value);
+            self.active_uncompressed_data_buffers
+                .insert(univariate_id, buffer);
         }
 
         None
     }
 
-    /// Remove the oldest [`FinishedSegment`] from the queue and return it. Return [`None`] if the
-    /// queue of [`FinishedSegments`](FinishedSegment) is empty.
-    pub(super) fn get_finished_segment(&mut self) -> Option<FinishedSegment> {
-        if let Some(finished_segment) = self.finished_queue.pop_front() {
-            // Add the memory size of the removed FinishedSegment back to the remaining bytes.
+    /// Remove the oldest finished [`UncompressedDataBuffer`] from the queue and return it. Returns
+    /// [`None`] if there are no finished [`UncompressedDataBuffers`](UncompressedDataBuffer) in the
+    /// queue.
+    pub(super) fn get_finished_data_buffer(&mut self) -> Option<Box<dyn UncompressedDataBuffer>> {
+        if let Some(uncompressed_data_buffer) = self.finished_uncompressed_data_buffers.pop_front()
+        {
+            // Add the memory size of the removed buffer back to the remaining bytes.
             self.uncompressed_remaining_memory_in_bytes +=
-                finished_segment.uncompressed_segment.get_memory_size();
+                uncompressed_data_buffer.get_memory_size();
 
-            Some(finished_segment)
+            Some(uncompressed_data_buffer)
         } else {
             None
         }
     }
 
     // TODO: Remove this when compression component is changed.
-    /// Compress the given full `segment_builder` and return the compressed and merged segment.
-    fn compress_full_segment(&mut self, mut segment_builder: SegmentBuilder) -> RecordBatch {
+    /// Compress the full [`UncompressedInMemoryDataBuffer`] given as `in_memory_data_buffer` and
+    /// return the resulting compressed and merged segments as a [`RecordBatch`].
```

Comment:
There are still a few instances of [`RecordBatch`] that does not have the `(datafusion::arrow::record_batch::RecordBatch)` link. Maybe fix this so all instances of [`RecordBatch`] have the link.

Reply:
`(datafusion::arrow::record_batch::RecordBatch)` is only needed if `RecordBatch` is not in scope. It was added in all of the locations were `cargo doc` emitted an error stating that it could not find a particular type. However, I found that `cargo doc` links to the local documentation for a crate instead of its online documentation, so to enable all links in the documentation it must be generated without `--no-deps`. Thus I have removed this flag from our GitHub Action workflow.